cmd/gardenlet/app/migration.go

```diff
@@ -36,6 +44,43 @@ func (g *garden) runMigrations(ctx context.Context, log logr.Logger, _ cluster.C
 	return nil
 }
 
+// TODO(timuthy): Remove after v1.96 was released.
+// updateGardenerResourceManager updates all GRM deployments in a seed cluster,
+// in order to roll out the Brotli compression support added in Gardener v1.96.
+func updateGardenerResourceManager(ctx context.Context, log logr.Logger, cl client.Client) error {
+	image, err := imagevector.ImageVector().FindImage(imagevector.ImageNameGardenerResourceManager)
+	if err != nil {
+		return err
+	}
+	image.WithOptionalTag(version.Get().GitVersion)
+
+	grmDeployments := &appsv1.DeploymentList{}
+	if err := cl.List(ctx, grmDeployments, client.MatchingLabels(map[string]string{"app": "gardener-resource-manager"})); err != nil {
+		return err
+	}
+
+	fns := make([]flow.TaskFn, 0, len(grmDeployments.Items))
+	for _, deployment := range grmDeployments.Items {
+		grmDeployment := deployment
+		if len(grmDeployment.Spec.Template.Spec.Containers) != 1 {
+			log.Info("Skipping migrated Gardener-Resource-Manager deployment because of missing container", "deployment", client.ObjectKeyFromObject(&grmDeployment))
+			continue
+		}
```

Comment:
The log message got me confused. It might be valid if there is no container, which would be surprising in a deployment. However, the more realistic case seems to be more than one container, e.g. due to injection of some side-car container. In that case the error message is leading in a completely wrong direction.

Could you please share your thoughts?